src/odemis/gui/cont/tools.py

```diff
@@ -258,8 +276,21 @@ def _add_button(self, cls, img_prefix, tooltip=None):
             f = wx.TOP | wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL
             b = 5
 
-        self.btn_sizer.Add(btn, border=b, flag=f)
-        self.btn_panel.Layout()
+        if self.orientation == (wx.VERTICAL | wx.HORIZONTAL):
+            if self.tool == TOOL_RULER:
+                self.btn_sizer.Add(btn, (0, 0), border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL)
+            elif self.tool == TOOL_LABEL:
+                self.btn_sizer.Add(btn, (1, 0), border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL)
+            elif self.tool == TOOL_POINT:
+                self.btn_sizer.Add(btn, (0, 1), border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL | wx.LEFT)
+            elif self.tool == TOOL_LINE:
+                self.btn_sizer.Add(btn, (1, 1), border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL | wx.LEFT)
+            elif self.tool == TOOL_ACT_ZOOM_FIT:
+                self.btn_sizer.Add(btn, (2, 1), border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL | wx.LEFT)
```

Comment:
There should be a else, which handles properly unknown tool ID. => Just pick a random place available.
I think you could try something like this:
```
# Ideal position for the known tools
pos = {TOOL_RULER: (0,0), TOOL_LABEL:(1,0)...}.get(self.tool)
# Unknown tool, or position already used => pick the first position available
if not pos or self.btn_sizer.FindItemAtPosition(pos):
   for j in range(8): # max 2x8
      for i in range(2):
        if not self.btn_sizer.FindItemAtPosition((i,j)):
           pos = (i,j)
           break
   else:
       raise ValueError("No more space in toolbar")
self.btn_sizer.Add(btn, pos, border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL)
```